### DISCUSSION TITLE
{{file_path:relative}} sometimes returns the *previously* active note

### DISCUSSION THREAD

**Discussion by ces3001** [Taitava/obsidian-shellcommands#379] [Debugging and testing]
I haven’t been able to determine when this happens, but often since the recent update to Obsidian (and/or Shell Commands), `{{file_path:relative}}` (and presumably without the `relative` parameter), will return the *previously* active note, not the currently active note.

**Comment by Taitava** [comment]
Thank you for your report. Does this happen when using the event [Switching the active pane](https://publish.obsidian.md/shellcommands/Events/Switching+the+active+pane)? Or in some other situation?

What version of Obsidian do you exactly have?
And what operating system?
I assume you are using SC 0.21.0?

I think there hasn't been any changes to events in SC during the last few months, so without better knowing yet, maybe it's a change in Obsidian. But I can't rule out a possibility that there could be a defect in SC. Will see.

**Comment by ces3001** [comment]
I did some testing and it look like Switching the active pane event detects the correct current page, but my personal use case that wasn’t getting the current page wasn’t using the events trigger,  but from an QuickAdd macro that calls the Shell Command.
Obsidian 1.5.3
macOS 14.1
I was on SC 0.20.1. I just upgraded and will see if it still occurs. 

On Jan 11, 2024, at 7:41 AM, Jarkko Linnanvirta ***@***.***> wrote:


Thank you for your report. Does this happen when using the event Switching the active pane <https://publish.obsidian.md/shellcommands/Events/Switching+the+active+pane>? Or in some other situation?

What version of Obsidian do you exactly have?
And what operating system?
I assume you are using SC 0.21.0?

I think there hasn't been any changes to events in SC during the last few months, so without better knowing yet, maybe it's a change in Obsidian. But I can't rule out a possibility that there could be a defect in SC. Will see.

**Reply by Taitava** [reply]
Ah, ok. Does your macro switch the focused note/pane right before calling the shell command?

I haven't used QuickAdd very much myself and it's been long since the last time. I took a very quick look at its settings now, and if I understood correctly, there doesn't seem to be a setting for a delay between macro actions, is there? I tried QuickAdd version `1.6.1`. Perhaps making the shell command to be called e.g. 200 ms later could solve the problem, if that would be possible in QuickAdd?

Edit 2024-06-04: QuickAdd does have an _Add wait command_ icon button when configuring a macro, which allows adding a delay.

**Comment by ces3001** [comment]
This is still an open issue. There is nothing before the Shell Command in the QuickAdd macro. The only thing that’s happening is that QuickAdd is being invoked with a hotkey to open a group of QuickAdd commands, which allows me to select one of them (different transcription services). If it’s first time (in a while) invoking it, Shell Commands gets the wrong open `file_path`, but once I’ve used it once, it will get it right from then on out. Adding a delay of 200ms makes no difference.

**Reply by Taitava** [reply]
I tried to create a macro in QuickAdd (on Windows and using PowerShell Core). I opened up the QuickAdd commands list via a hotkey, selected my macro and it ran correctly. I tried it multiple times and also restarted Obsidian in between. So I can't really do much research on it before I'm able to reproduce it.

@tyf2018 seems to have the same problem. Quote from #413:
> 1. Open File A
> 2. Switch the current tab to File B
> 3. Run the command：`C:\totalcmd\Tools\Emeditor\EmEditor.exe "{{file_path:absolute}}" /l {{caret_position}}`
> 4. The expected result should be opening File B in EmEditor. The actual result: most of the time it works correctly, but occasionally it opens File A instead of File B.
> 
> Possible Reason: My vault is quite large, with over ten thousand notes, which may be causing delays in data retrieval.

The use-case for @tyf2018 is a bit different though, as they don't seem to be using QuickAdd (or any other plugin in addition to SC) to run the SC plugin's command. This is valuable information, as it suggests that the problem should be reproducible without other plugins. This should simplify the scenario, hopefully.

@ces3001 do you happen to have a large vault, similarly as @tyf2018 has over ten thousand notes?

Next things I'd like to ask:
- Can you please record a video of the problem occurring? Maybe this could show me some minor details of the scenario when the bug occurs - something I might have missed so far. You can blur parts of the video if you don't want to show note contents etc.
- Can you please also share the _Shell commands_ plugin's settings file (`.obsidian/plugins/obsidian-shellcommands/data.json`), if it doesn't contain private content? Maybe seeing the overall configuration of the plugin can shed some light on this problem. E.g. if you happen to have multiple shell commands, perhaps a bug could make them interfere with each others in theory.
- And speaking of settings files, perhaps QuickAdd's settings file would be good to see, too, please.

If you don't like sharing these things, it's okay, we can try to find other ways to research this problem. 👍
